translator.py

```diff
@@ -169,23 +179,32 @@ def _translation_loop(self):
                         # No multicast address currently allocated for this client. Allocate one.
                         mcast_addr = self._alloc_mcast_addr(src_addr)
                         if mcast_addr is not None:
-                            # Publish the new stream on the Multicast Menu if we successfully allocated a multicast
-                            # address. We'll let a worker thread handle the communication with the multicast menu
-                            # s.t. the I/O does not block translation of any ongoing streams (i.e., it is important we
-                            # do not block the thread that runs the translation loop).
+                            # Publish the new stream on Multicast Menu if we successfully allocated a multicast
+                            # address and have a UID to communcate with Multicast Menu. We'll let a worker thread 
+                            # handle the communication with Multicast Menu s.t. the I/O does not block translation 
+                            # of any ongoing streams (i.e., it is important we do not block the thread that runs 
+                            # the translation loop).
                             desc = f'Translated stream originating from {src_addr[0]}'
-                            self._mcastmenu_threadpool.submit(self._add_to_multicast_menu, mcast_addr,
-                                                              constants.MULTICASTMENU_EMAIL, desc)
+                            if self._mm_uid is not None:
+                                self._mcastmenu_threadpool.submit(self._add_to_multicast_menu, mcast_addr, src_addr)
                     if mcast_addr is not None:
                         # Forward the payload to the multicast address allocated for this client.
                         self._mcast_sckt.sendto(payload, (str(mcast_addr), self._mcast_dst_port))
+                        # Update Activity Information Base
+                        self._atb[src_addr] = datetime.now()
```

Comment:
I think we need to add locking around this line as it is a mutating operation and you run `self._identify_inactive_streams` on a different thread, which also manipulates this dict